### PR TITLE
buffer: fix per-route buffer disable behavior

### PR DESCRIPTION
**Pull request by mattklein123**
Also remove more dead code from the timer removal.

Fixes https://github.com/envoyproxy/envoy/issues/5513

*Risk Level*: Low
*Testing*: Fixed existing tests
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@vladimir-spotify @auni53 @rodaine

**Comment by vladimir-spotify** [comment]
hey, thank you!
can confirm - it works for us now.

File: test/extensions/filters/http/buffer/buffer_filter_test.cc [hunk 1]
```diff
@@ -33,7 +33,7 @@ class BufferFilterTest : public testing::Test {
   BufferFilterConfigSharedPtr setupConfig() {
     envoy::config::filter::http::buffer::v2::Buffer proto_config;
     proto_config.mutable_max_request_bytes()->set_value(1024 * 1024);
-    return std::make_shared<BufferFilterConfig>(BufferFilterConfig(proto_config, "test", store_));
+    return std::make_shared<BufferFilterConfig>(BufferFilterConfig(proto_config));
```
Comment (jmarantz) [inline test/extensions/filters/http/buffer/buffer_filter_test.cc]: doesn't it work to just write:
```
return std::make_shared<BufferFilterConfig>(proto_config);
```
?

Comment (mattklein123) [reply test/extensions/filters/http/buffer/buffer_filter_test.cc]: Done

**Comment by mattklein123** [comment]
@jmarantz updated

**Pull request merged**